src/ConnectedMode/Binding/ImportBeforeInstallTrigger.cs

```diff
@@ -29,21 +31,38 @@ namespace SonarLint.VisualStudio.ConnectedMode.Binding
     [PartCreationPolicy(CreationPolicy.Shared)]
     internal class ImportBeforeInstallTrigger
     {
-        private readonly IConfigurationProvider configurationProvider;
+        private readonly IActiveSolutionBoundTracker activeSolutionBoundTracker;
```

Comment:
The class probable needs to be `sealed` to prevent a different code smell about the implementation of `IDisposable`.